### PR TITLE
Permit ravel('A') for contig device arrays in CUDA target

### LINKED ISSUE #4827: Trouble using `.ravel("A")` on contiguous CUDA DeviceNDArrays
## Reporting a bug

- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

When trying to use `.ravel("A")` on a device array that is C-contiguous (though have seen the same with F-contiguous device arrays), it complains that the array is not contiguous. However `"A"` (at least for NumPy) means the array could be C-contiguous or F-contiguous and still be raveled without a copy. Would figure this would be the case for Numba as well, but it appears not. Please let me know if I'm just doing something silly though.

```python
In [1]: import numba.cuda                                                       

In [2]: a = numba.cuda.device_array((10, 12))                                   

In [3]: a.ravel("C")                                                            
Out[3]: <numba.cuda.cudadrv.devicearray.DeviceNDArray at 0x7f4a88394710>

In [4]: a.ravel("A")                                                            
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-4-65fd138a7dd9> in <module>
----> 1 a.ravel("A")

~/miniconda/envs/cupy37/lib/python3.7/site-packages/numba/cuda/cudadrv/devicearray.py in ravel(self, order, stream)
    483         stream = self._default_stream(stream)
    484         cls = type(self)
--> 485         newarr, extents = self._dummy.ravel(order=order)
    486 
    487         if extents == [self._dummy.extent]:

~/miniconda/envs/cupy37/lib/python3.7/site-packages/numba/dummyarray.py in ravel(self, order)
    358 
    359         else:
--> 360             raise NotImplementedError("ravel on non-contiguous array")
    361 
    362 

NotImplementedError: ravel on non-contiguous array
```

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

Fixes #4827

### CODE REVIEW

**Comment by stuartarchibald** [comment]
I think this is the fastest fix for #4827, though it might be worth investigating the noted https://github.com/numba/numba/issues/4827#issuecomment-553053749

File: numba/cuda/tests/cudadrv/test_array_attr.py [hunk 1]
```diff
@@ -43,22 +43,33 @@ def test_contigous_4d(self):
...
         self.assertTrue(flat.ndim == 1)
         self.assertTrue(np.all(expect == flat))
 
+        # explicit order kwarg
+        for order in 'CA':
+            expect = reshaped.ravel(order=order)
+            dary = cuda.to_device(reshaped)
+            dflat = dary.ravel(order=order)
+            flat = dflat.copy_to_host()
+            self.assertTrue(flat.ndim == 1)
```
Comment (sklam) [inline numba/cuda/tests/cudadrv/test_array_attr.py]: better use `assertEqual`

File: numba/cuda/tests/cudadrv/test_array_attr.py [hunk 2]
```diff
@@ -43,22 +43,33 @@ def test_contigous_4d(self):
...
+            self.assertTrue(np.all(expect == flat))
```
Comment (sklam) [inline numba/cuda/tests/cudadrv/test_array_attr.py]: Better use `assertPreciseEqual` or `np.testing.assert...`

File: numba/cuda/tests/cudadrv/test_array_attr.py [hunk 3]
```diff
@@ -43,22 +43,33 @@ def test_contigous_4d(self):
...
-        dflat = dary.ravel(order='F')
-        flat = dflat.copy_to_host()
-        self.assertTrue(flat.ndim == 1)
-        self.assertTrue(np.all(expect == flat))
+        for order in 'FA':
+            expect = reshaped.ravel(order=order)
+            dary = cuda.to_device(reshaped)
+            dflat = dary.ravel(order=order)
+            flat = dflat.copy_to_host()
+            self.assertTrue(flat.ndim == 1)
```
Comment (sklam) [inline numba/cuda/tests/cudadrv/test_array_attr.py]: same here

File: numba/cuda/tests/cudadrv/test_array_attr.py [hunk 4]
```diff
@@ -43,22 +43,33 @@ def test_contigous_4d(self):
...
+            self.assertTrue(np.all(expect == flat))
```
Comment (sklam) [inline numba/cuda/tests/cudadrv/test_array_attr.py]: and here

File: numba/cuda/tests/cudadrv/test_array_attr.py (hunk 1 above)
Comment (stuartarchibald) [reply numba/cuda/tests/cudadrv/test_array_attr.py]: fixed all these (and some more in the same file)

**Comment by seibert** [comment]
running internal CI: `numba_65`

**Comment by seibert** [comment]
This has passed enough internal configurations (we have a bit of a backlog) for me to feel confident this is fine.

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**